core/src/trezor/workflow.py

```diff
@@ -76,12 +95,21 @@ def close_default() -> None:
 
 
 def _finalize_default(task: loop.Task, value: Any) -> None:
+    """Finalizer for the default task. Cleans up globals and restarts the default
+    in case no other task is running."""
     global default_task
 
     if default_task is task:
         if __debug__:
-            log.debug(__name__, "default closed")
+            log.debug(__name__, "default closed: %s", task)
         default_task = None
+
+        if not tasks and default_constructor:
+            # No registered workflows are running and we are in the default task
+            # finalizer, so when this function finished, nothing will be running.
+            # We must schedule a new instance of the default now.
+            start_default(default_constructor)
```

Comment:
Do we want to add this here to have it a bit more under control?

```python
if not tasks and not default_constructor:
   raise RuntimeError  # we need to have a default workflow
```
? And remove the loop from main.py because the reboot is unwanted at the moment. More of a question.

Reply:
that could be useful, yes.

i'm not sure about the role the loop has in main.py. Under normal conditions `loop.run` should never exit: even if we raise here and exit the default tasks, `handle_session` will still be running.
Maybe main.py should end with `utils.halt` to make visible situations where everything crashes?

Reply:
Yes I agree. I think it's better being explicit in these situations to know what is happening